src/CredentialsLoader.php

```diff
@@ -169,4 +169,13 @@ public function updateMetadata(
     $metadata_copy[self::AUTH_METADATA_KEY] = array('Bearer ' . $result['access_token']);
     return $metadata_copy;
   }
+
+  /**
+   * Gets the underlying `OAuth2` instance used to conduct authorization.
+   *
+   * @return OAuth2
+   */
+  public function getAuth() {
```

Comment:
For my use case, I don’t want to and should not use the middleware classes. I also realized my statement above about `fetchAuthToken()` being an implicit contract is incorrect since I forgot that cred loader impls `FetchAuthTokenInterface`. I also realized that `getExpiresAt()` is actually the method I want, since it calculates the expiry time whereas `getExpiresIn()` doesn’t (it should probably be documented that it’s a snapshot of the token metadata when the token was first received). There’s also a bug I fixed with `getExpiresAt()` in `updateToken()` being overriden to `null`.

Anyways, what I propose for now is we implement `getExpiresAt()` on the `FetchAuthTokenInterface` interface, since in `AuthTokenMIddleware’s` `fetchToken()` method, you’re going to need to get the expiry time of the token if you want to set it in the cache. I know the TODO’s say that this should be read from the access token array returned from `fetchAuthToken()` and set in the cache, but that means you’d be calculating the expiry time again, since the logic is already available in the `OAuth2` class. So how does exposing `getExpiresAt()` on `FetchAuthTokenInterface` sound? All the subclasses will just delegate this to the underlying `OAuth2` instance anyways. I can do this in another pull request (or this one) - up to you.

This way we don’t need to expose the underlying `OAuth2` instance on cred loader.

Also, I think the larger issue here is that we don’t have an access token class. The access token itself is not stored in the `OAuth2` object and only stored if you provide a cache. But the metadata (like expiry time) for the token is stored in the `OAuth2` object - so it’s decoupled. An access token class should probably be created and this metadata moved there?